### PR TITLE
Add Xcode auto-generated XCSchemes to .gitignore

### PR DESCRIPTION
**Pull request by rdas-ditto**
Excluded the auto-generated XCSchemes from Swift Package Manager under .swiftpm/xcode/xcshareddata/xcschemes/ by adding them to .gitignore. 

This change prevents unnecessary files from being tracked in the repository, as these schemes are auto-generated by Xcode and can be safely ignored to keep the version control system clean.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,6 +9,9 @@ xcuserdata/
 *.xcscmblueprint
 *.xccheckout
 
+# Ignore Xcode scheme files generated by SwiftPM
+.swiftpm/xcode/xcshareddata/xcschemes/
```
Comment (bplattenburg) [inline .gitignore:13]: I was looking at some other `.gitignore` changes, and I noticed there's a whole SPM section commented out farther down - Maybe it makes the most sense to uncomment all of that and cleanup whats checked in today?

That could capture all of `.swiftpm` as well as quite a few other autogenerated things

**Comment by rdas-ditto** [comment]
Good catch, I've updated the .gitignore, and also removed the existing tracked files (xcscheme, and also Package.resolved)

File: .gitignore [hunk 2]
```diff
@@ -46,7 +46,7 @@ playground.xcworkspace
 #
 # Xcode automatically generates this directory with a .xcworkspacedata file and xcuserdata
 # hence it is not needed unless you have added a package configuration file to your project
-# .swiftpm
+.swiftpm
```
Comment (bplattenburg) [inline .gitignore:49]: I think we should also uncomment lines 42, 43, and 44 above.

Comment (rdas-ditto) [reply .gitignore:49]: Done 👍

**Review by bplattenburg — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -39,14 +39,14 @@ playground.xcworkspace
...
 # Add this line if you want to avoid checking in source code from Swift Package Manager dependencies.
-# Packages/
-# Package.pins
-# Package.resolved
-# *.xcodeproj
-#
+Packages/
+Package.pins
+Package.resolved
+*.xcodeproj
```
Comment (bplattenburg) [inline .gitignore:45]: I think we actually need this to stay since we have the Tools App project in here

Comment (rdas-ditto) [reply .gitignore:45]: Ok re-added them, except `Package.resolved` and `.pins` which are generated by SPM so don't need to be tracked in this case.

**Comment by rdas-ditto** [comment]
This is ready to review, and merge.

**Review by bplattenburg — APPROVED** [review]

**Pull request merged**